### ISSUE TITLE
[FP] Optional before required parameter

### ISSUE THREAD

**Issue by orklah** [kalessil/phpinspectionsea#1150]
| Subject        | Details                                                                       |
| :------------- | :---------------------------------------------------------------------------- |
| Issue type     | False-Positive                         |
| Plugin         | Php Inspections (EA Extended)  |
| Language level | PHP5.*, PHP 7.0                       |

Current behaviour (description/screenshot:)
--
Consider the following code:
```php
test(new stdClass(), 'param');
test(null, 'param');
function test(stdClass $obj, $mandatory_param){}
```
[3v4l link](https://3v4l.org/VtYgY)

There is an error because it is impossible to pass null to a typed param.

Before PHP 7.1 introduced nullable arguments, there was only 2 possibility to fix the issue:
- Change the signature to function test(stdClass $obj = null, $mandatory_param){}
- Change the signature to function test($obj, $mandatory_param){}

Both options have downside but my preferred is the first one. However, the "Optional before required parameter" inspection is now displayed.

Expected behaviour
--
If an optional parameter has a type hint and a default value to = null, don't display the inspection for versions prior to PHP 7.1

**Comment by kalessil** [comment]
Checked, the "PHP | Probable bugs | Optional before required parameter" inspection is from IDE itself. Perhaps it's something known already: https://youtrack.jetbrains.com/issues

**Comment by orklah** [comment]
My bad, I thought that was one of yours!

**Comment by kalessil** [comment]
No worries, I are getting similar ticket from time to time =)
